### PR TITLE
[dev] Update EKS Cluster targets to 1.24 - 1.27

### PR DESCRIPTION
**Pull request by Kausik-A**
**Description:** 

Update EKS Cluster targets to 1.24 - 1.27

**Link to tracking Issue:** https://github.com/aws-observability/aws-otel-collector/issues/2427

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/config/testcases.json [hunk 1]
```diff
@@ -4,35 +4,28 @@
...
 				{
-					"name": "collector-ci-amd64-1-22",
-					"region": "us-west-2",
-					"excluded_tests": [
-						"containerinsights_eks_containerd"
-					]
-				},
-				{
-					"name": "collector-ci-amd64-1-23",
+					"name": "collector-ci-amd64-1-24",
```
Comment (bryan-aguilar) [inline .github/config/testcases.json:7]: We should only be testing new clusters on the dev branch. Not clusters we are already testing against.

**Review by bryan-aguilar — CHANGES_REQUESTED** [review]
CI.yaml platform targets needs to be updated. https://github.com/aws-observability/aws-otel-collector/blob/3fe9d45793e989cdc28d7343444e3c5f3bf833cb/.github/workflows/CI.yml#L52

File: .github/config/testcases.json (hunk 1 above)
Comment (Kausik-A) [reply .github/config/testcases.json]: Makes sense. I have updated the testcases

File: .github/workflows/CI.yml [hunk 2]
```diff
@@ -49,7 +49,7 @@ env:
   GITHB_RUN_ID: ${{ github.run_id }}
   DDB_TABLE_NAME: BatchTestCache
   MAX_JOBS: 110
-  BATCH_INCLUDED_SERVICES: ECS,EC2
+  BATCH_INCLUDED_SERVICES: ECS,EC2,EKS,EKS_ARM64
```
Comment (bryan-aguilar) [inline .github/workflows/CI.yml:52]: Farge
```suggestion
  BATCH_INCLUDED_SERVICES: ECS,EC2,EKS,EKS_ARM64,EKS_FARGATE
```

**Review by bryan-aguilar — APPROVED** [review]

**Pull request merged**